test-apps/ui-test-app/src/frontend/appui/imodelindex/IModelIndex.tsx

```diff
@@ -88,41 +96,38 @@ export class IModelIndex extends React.Component<IModelIndexProps, IModelIndexSt
     IModelIndex._categories = IModelIndex._categories.filter((_category: Category) => _category.label !== _label);
   }
 
-  /* retrieves the iModel thumbnail. */
-  private async startRetrieveThumbnail(iTwinId: string, iModelId: string) {
-    const hubFrontend = new IModelHubFrontend();
-    const _thumbnail = await hubFrontend.hubClient.thumbnails.download((await IModelApp.getAccessToken())!, iModelId, { iTwinId, size: "Small" });
-    this.setState({ thumbnail: _thumbnail });
-  }
-
   /* retrieve version information */
   private async startRetrieveIModelInfo() {
-    const hubClient: IModelClient = new IModelHubClient();
-    const iTwinId = this.props.iModelConnection.iTwinId!;
     const iModelId = this.props.iModelConnection.iModelId!;
     const accessToken = await IModelApp.getAccessToken();
 
-    /* get the iModel name */
-    const imodels = await hubClient.iModels.get(accessToken, iTwinId, new IModelQuery().byId(iModelId));
+    if (!SampleAppIModelApp.hubClient)
+      return;
 
-    /* get the top named version */
-    const _versions: Version[] = await hubClient.versions.get(accessToken, iModelId, new VersionQuery().top(1));
-
-    /* determine if the version is up-to-date */
-    const changeSetId = this.props.iModelConnection.changeset.id;
-    const _upToDate = (_versions.length > 0 && _versions[0].changeSetId === changeSetId);
+    const imodel = await SampleAppIModelApp.hubClient?.iModels.getSingle({
+      iModelId,
+      authorization: async () => toAuthorization(accessToken),
+    });
 
     /* get the version name */
-    let currentVersions: Version[] = [];
+    const currentVersions: NamedVersion[] = [];
     let _versionName = "";
     try {
-      currentVersions = await hubClient.versions.get(accessToken, iModelId, new VersionQuery().byChangeSet(changeSetId));
-      _versionName = (currentVersions.length === 1) ? currentVersions[0].name! : "Version name not found!";
+      for await (const ver of SampleAppIModelApp.hubClient?.namedVersions.getRepresentationList({
+        urlParams: {
+          $top: 1,
```

Comment:
The previous implementation seems to query version by changeset id and the new one queries all, is this intentional? 

Also if you expect the collection to only contain a single item (I am assuming this because of `[0]` usage) specifying the `$top` param is not enough. It only specifies page size and not how many entities to return in a collection so if you iterate over all items multiple requests may be sent. If you want to send a single request that returns a single item in page either stop the iteration after the first item or use [`take` function](https://github.com/iTwin/imodels-clients/blob/513b8f6954b934983e361ae388d26a6e1cba8a1c/clients/imodels-client-management/src/base/PagedCollectionGenerator.ts#L40).

Reply:
I didn't see an easy way to get query for a specific NamedVersion by changesetId so I'm iterating. I'll take another look at the API to see if I can do it a better way.

In the meantime, just updated to break after the first iteration.